primitives/trie/src/recorder.rs

```diff
@@ -135,6 +158,69 @@ impl<H: Hasher> Recorder<H> {
 		mem::take(&mut *self.inner.lock());
 		self.encoded_size_estimation.store(0, Ordering::Relaxed);
 	}
+
+	/// Stat a new transaction.
+	pub fn start_transaction(&self) {
+		let mut inner = self.inner.lock();
+		inner.accessed_nodes_transactions.push(Default::default());
+		inner.recorded_keys_transactions.push(Default::default());
```

Comment:
I mean both walk in lockstep, but there is no assumption that both have the same length? We never access anything from one to the other assuming that the same index exists there as well.

Reply:
I meant silent assumption - at the end both vecs should have the length of number of transactions, it should never happen that one contains more then other. Assert would be more expressive way to indicate this bound. It would also help if some modifies the code (e.g. return early from `rollback/commit_transaction`).

Anyway, feel free to ignore it.

Reply:
Done